
Async connection drain was removed* in v23, and does not build here
with clang-16. Disable it.

* https://github.com/ClickHouse/ClickHouse/pull/47486

diff --git a/src/QueryPipeline/RemoteQueryExecutor.cpp b/src/QueryPipeline/RemoteQueryExecutor.cpp
index 44e844f..534074c 100644
--- a/src/QueryPipeline/RemoteQueryExecutor.cpp
+++ b/src/QueryPipeline/RemoteQueryExecutor.cpp
@@ -497,6 +497,7 @@ void RemoteQueryExecutor::finish(std::unique_ptr<ReadContext> * read_context)
     /// Send the request to abort the execution of the request, if not already sent.
     tryCancel("Cancelling query because enough data has been read", read_context);
 
+#if 0
     if (context->getSettingsRef().drain_timeout != Poco::Timespan(-1000000))
     {
         auto connections_left = ConnectionCollector::enqueueConnectionCleanup(pool, connections);
@@ -509,6 +510,7 @@ void RemoteQueryExecutor::finish(std::unique_ptr<ReadContext> * read_context)
         }
     }
     else
+#endif
     {
         /// Drain connections synchronously without suppressing errors.
         CurrentMetrics::Increment metric_increment(CurrentMetrics::ActiveSyncDrainedConnections);
